runtime/near-vm-runner/tests/current_vm_does_not_crash/main.rs

```diff
@@ -0,0 +1,47 @@
+use arbitrary::Arbitrary;
+use bolero::check;
+use near_primitives::contract::ContractCode;
+use near_primitives::runtime::fees::RuntimeFeesConfig;
+use near_primitives::version::PROTOCOL_VERSION;
+use near_vm_logic::mocks::mock_external::MockedExternal;
+use near_vm_logic::VMConfig;
+use near_vm_runner::internal::VMKind;
+use near_vm_runner::VMResult;
+
+#[path = "../common/mod.rs"]
+mod common;
```

Comment:
Heh, not a huge fan of including a file into two CUs, for all the usual reasons: confused IDEs, more compiling, spurious unused warnings, straying away from "convention over configuration". 

I don't think this is blocking, but it'd be great to work with bolero to fix this. 


One  thing we can do here is to just put this stuff into `lib.rs`: `find_entry_point` and `create_context` definitely belong there. 

I *think* something like `impl Arbitrary for ContractCode` is also potentially fine for `lib.rs`? That'd require making arbitrary and wasm-smith non-dev dependencies, but I think that might be OK. 

<hr/>

This is what we can do on our side. But I think a more fundamental fix is maybe warranted on the bolero's side (cc @camshaft). The core problem here is that the two fuzz tests are separate binaries. I think this is pretty horrible, as it means 2x linking work. Basically, https://matklad.github.io/2021/02/27/delete-cargo-integration-tests.html. 


So I am wondering if bolero could do "several tests in one binary" thing? Basically, something like this: 

```
fn vm_does_not_crash(data: &[u8]) { ... }
fn wasmer2_and_wasmtime_agree(data: &[u8]) { ... }

bolero::main![
    vm_does_not_crash,
    wasmer2_and_wasmtime_agree,
];
```

?

Reply:
Yeah if you want to go with that approach, you can use the libtest harness instead:

```rust
#[test]
fn first() {
    bolero::check!()...
}

#[test]
fn second() {
    bolero::check!()...
}
```

Note that bolero tests can also be compiled as part of the main unit test binary as well so you can get rid of all of the integration binaries. (See for example https://github.com/aws/s2n-quic/blob/e92f55bb7fcb9fdf184f381d631c450817455d2c/quic/s2n-quic-core/src/slice.rs#L158)

Reply:
@camshaft Can bolero run eg. libfuzzer on tests that are using the libtest harness? We can probably live without bolero::check! getting artifacts from `corpus/` and `artifacts/` (by manually upgrading these to unit tests when we'll have failure in the fuzzer), but we do need the ability to run libfuzzer on the fuzz targets, and I've seen no documentation about how to make this work?

(If that's supposed to work, and not just the "run 1024 times the test", then it's probably the best option for us!)

@matklad I don't think this is wrong on the cargo-bolero side, IMO this is something that is wrong on the cargo side in that it AFAICT provides no way to have shared code that's only available for integration tests unless the integration test is just one file. IOW, https://doc.rust-lang.org/book/ch11-03-test-organization.html#submodules-in-integration-tests does work but not as soon as the integration test is set to `tests/[testname]/main.rs`, and this feels wrong to me, hence my trying to catch that up with #[path]. If you know of a better way to recover this documented behavior from integration tests that deserve their full folder I'd love to hear it! I'm also considering opening a cargo issue there to get other people's opinions. WDYT?

As for hoisting all these things into `lib.rs`, I'm conflicted because it means that now everyone will have to care about a few functions that are designed only for tests and should never be used for anything else. We can #[doc(hidden)] them I guess (though we can't make dependencies disappear), but to me it feels wronger than having one file compiled in two compilation units… though I guess not working on rust-analyzer is what makes me feel the issues of multiple compilation units less :sweat_smile:

Reply:
> Can bolero run eg. libfuzzer on tests that are using the libtest harness? 

Yep! You just refer to the path of your test harness when running cargo-bolero:

```sh
$ cargo bolero test path::to::test -p my-crate
```

Reply:
Ooooh this looks great, thank you!